lib/http.js

```diff
@@ -0,0 +1,45 @@
+import { getLatestMetrics, followMetrics } from '../lib/metrics.js'
+import http from 'node:http'
+import { once } from 'node:events'
+
+const asyncHandler = async (req, res) => {
+  const url = new URL(req.url, 'http://localhost/')
+
+  if (url.pathname === '/metrics') {
+    res.setHeader('Content-Type', 'application/json')
+    if (url.searchParams.has('follow')) {
+      const controller = new AbortController()
+      const { signal } = controller
+      await Promise.all([
+        (async () => {
+          try {
+            await once(req, 'close')
+          } catch {
+          } finally {
+            controller.abort()
+          }
+        })(),
+        (async () => {
+          try {
+            for await (const line of followMetrics({ signal })) {
+              res.write(`${line}\n`)
+            }
+          } catch {}
```

Comment:
I think you should end the response on error (e.g. when the `followMetrics` iterator throws). Otherwise, the response stream remains open forever.